doc/NAC/PnacSonicHLD.md

```diff
@@ -0,0 +1,374 @@
+                    "trap_priority":"4",                    
+                    "queue": "4",                    
+                    "meter_type":"packets",                    
+                    "mode":"sr_tcm",                    
+                    "cir":"600",                    
+                    "cbs":"600",                    
+                    "red_action":"drop"                    
+            },            
+            "queue4_group3": {            
+                    "trap_action":"trap",                    
+                    "trap_priority":"4",                    
+                    "queue": "4"                    
+            },            
+            "queue1_group1": {            
+                    "trap_action":"trap",                    
+                    "trap_priority":"1",                    
+                    "queue": "1",                    
+                    "meter_type":"packets",                    
+                    "mode":"sr_tcm",                    
+                    "cir":"6000",                    
+                    "cbs":"6000",                    
+                    "red_action":"drop"                    
+            },
+            "queue1_group2": {            
+                    "trap_action":"trap",                    
+                    "trap_priority":"1",                    
+                    "queue": "1",                    
+                    "meter_type":"packets",                    
+                    "mode":"sr_tcm",                    
+                    "cir":"600",                    
+                    "cbs":"600",                    
+                    "red_action":"drop"                    
+            },
+            "queue2_group1": {            
+                    "cbs": "1000",                    
+                    "cir": "1000",                    
+                    "genetlink_mcgrp_name": "packets",                    
+                    "genetlink_name": "psample",                    
+                    "meter_type": "packets",                    
+                    "mode": "sr_tcm",                    
+                    "queue": "2",                    
+                    "red_action": "drop",                    
+                    "trap_action": "trap",                    
+                    "trap_priority": "1"                     
+             }             
+    },    
+    "COPP_TRAP": {    
+            "nac": {            
+                    "trap_ids": "eapol",                    
+                    "trap_group": "queue4_group3"                    
+            }            
+    }    
+}
+```
+After configurating “***NAC Adminstate***” to “***up***” for a given interface, update in config_db triggers 1following actions
+Update learn_mode for the specific interface in config_db to “***SAI_BRIDGE_PORT_FDB_LEARNING_MODE_DROP***”
```

Comment:
Is CONFIG_DB being used to change the learning mode of the port? Isn't this change an operational change w.r.t PNAC ?  If user saves config while authentication is in progress (after learn mode has been set to DROP), are there repurcussions?

Reply:
1. Yes we are using config DB to trigger changes in the learning mode .
2. To answer your second question (by considering your scenario) - In the current behavior, the correct mode of DROP, FWD shall be reflected at the very next instance of reauthentication process.